ament_package/package.py

```diff
@@ -98,9 +112,13 @@ def validate(self):
 
         if not self.name:
             errors.append('Package name must not be empty')
-        # only allow lower case alphanummeric characters and underscores
-        # must start with an alphabetic character
-        if not re.match('^[a-z][a-z0-9_]*$', self.name):
+        valid_package_name_regexp = '^[a-z][a-z0-9_]*$'
+        # must start with an alphabetic character only allow lower case
+        # alphanummeric characters and underscores in catkin or ament packages.
+        # Dashes are allowed for other build_types.
+        if not re.match('^(ament[a-z0-9_]*|catkin)$', self.get_build_type()):
```

Comment:
I wouldn't use a regex for this since the logic doesn't require one: `if build_type != 'catkin' and not build_type.startswith('ament'):`